cmd/md5sum/md5sum.go

```diff
@@ -27,9 +28,22 @@ By default, the hash is requested from the remote.  If MD5 is
 not supported by the remote, no hash will be returned.  With the
 download flag, the file will be downloaded from the remote and
 hashed locally enabling MD5 for any remote.
+
+This command can also hash data received on standard input (stdin),
+by not passing a remote:path, or by passing a hyphen as remote:path
+when there is data to read (if not, the hypen will be treated literaly,
+as a relative path).
 `,
-	Run: func(command *cobra.Command, args []string) {
-		cmd.CheckArgs(1, 1, command, args)
+	RunE: func(command *cobra.Command, args []string) error {
+		cmd.CheckArgs(0, 1, command, args)
+		if len(args) == 0 {
+			return hashsum.HashStream(hash.MD5, hashsum.OutputBase64, os.Stdin)
+		}
+		if args[0] == "-" {
+			if fi, _ := os.Stdin.Stat(); fi.Mode()&os.ModeCharDevice == 0 {
```

Comment:
What is the purpose of the `fi.Mode()&os.ModeCharDevice == 0` check? If the user passes `-` don't we use stdin regardless?

Reply:
It checks if there is data on stdin. Same thing was done by obscure command. It means you can have a file named `-` and refer to it with relative path: `rclone md5sum "-"`. If you send it data it will be hashed instead of the file: `echo "something" | rclone md5sum -`

Reply:
I guess this would stop you running `rclone md5sum - < /dev/ttyUSB1` but that is probably pretty unlikely!

We'll see if we get any bug reports (I doubt we will!)

Reply:
Oh, that's not something I'm too familiar with.. But you could drop the hypen to achieve this, just use: `rclone md5sum < /dev/ttyUSB1`, as the hypen is optional (same as with the gnu tools). If I change to always treat `-` as stdin, we have the equally unlikely case that someone will have a relative path reference to something named "-"? 50-50 call? If you are 51% on changing it I will do it right away, if not I'll keep it as is  🤷‍♂️

Reply:
Let's keep it as is.